ci_framework/roles/libvirt_manager/tasks/attach_interface.yml

```diff
@@ -14,21 +14,39 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
-
 # This task requires
 #   vm_name         Domain name to which the interface needs to be attached.
-#   interface_name  source of the device
-#   interface_type  bridge or network
+#   network
+#     name          Name of the network bridge
+#     cidr          network notation
+#     mtu           MTU applied on the interface.
+#     static_ip     boolean if static address is required.
+
+
+- name: Generate a random MAC address
+  ansible.builtin.set_fact:
+    _lm_mac_address: "{{ '0A:02' | community.general.random_mac }}"
+
+- name: Reserve an IP address
+  when: network.static_ip | bool
```

Comment:
ah, nice catch. Will see to support that one as well. I thought about it, then just went for "meh, let's fix all IPs".

The fact I'm doing the reverse from within libvirt_manager (i.e. create the VM with associated NICs, then get the NICs, then fix IP, ...) is making things a liiiitle bit more complicated, but I'll figure out how to filter and pass the right data to that tasks file.